src/common/crypto/mbedtls/driverwrapper.cpp

```diff
@@ -298,69 +319,71 @@ aos::RetWithError<KeyInfo> AosPsaAddKey(const aos::crypto::PrivateKeyItf& privKe
     aos::LockGuard lock(sMutex);
 
     for (psa_key_id_t keyID = MBEDTLS_PSA_KEY_ID_BUILTIN_MIN; keyID <= MBEDTLS_PSA_KEY_ID_BUILTIN_MAX; ++keyID) {
-        bool found {};
+        if (IsKeyInBuiltinList(keyID)) {
+            continue;
+        }
 
-        for (auto& key : sBuiltinKeys) {
-            if (key.mKeyID == keyID) {
-                found = true;
+        auto [slotNumber, err] = FindFreeSlot();
 
-                break;
+        if (!err.IsNone()) {
+            if (err = sBuiltinKeys.PushBack(KeyDescription()); !err.IsNone()) {
```

Comment:
We can initialize array with max size. But it requires to add this type of constructor to StaticArray.

Reply:
Done